include/deal.II/matrix_free/vector_access_internal.h

```diff
@@ -24,9 +24,7 @@
 #include <deal.II/matrix_free/matrix_free.h>
 #include <deal.II/matrix_free/type_traits.h>
 
-#ifdef DEBUG
-#  include <boost/algorithm/string/join.hpp>
-#endif
+#include <boost/algorithm/string/join.hpp>
```

Comment:
In this file, there is some code that was only used inside `#ifdef DEBUG ... #endif`, and so this header file here was only included in that case. Now the code is inside `if constexpr` and the compiler complains about that code in release mode because some functions called there aren't declared -- so I need to unconditionally `#include` this file here.